File: test_api.js
const http = require('http');

const data = JSON.stringify({
  topic: 'Yoga',
  mode: 'GENERAL',
  strategy_mode: 'incubation',
  category: 'Fitness Equipment'
});

const options = {
  hostname: 'localhost',
  port: 3000,
  path: '/api/agent-forum',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
};

const req = http.request(options, (res) => {
  let responseData = '';
  res.on('data', (chunk) => {
    responseData += chunk;
  });
  res.on('end', () => {
    console.log('Status Code:', res.statusCode);
    console.log('Response Body:');
    console.log(responseData);
  });
});

req.on('error', (error) => {
  console.error('Request error:', error.message);
});

req.write(data);
req.end();

File: test_api.py
import requests
import json

print("Testing FastAPI server...")

# Test 1: Health check
try:
    response = requests.get('http://localhost:8000/')
    print(f"Health check status code: {response.status_code}")
    print(f"Health check response: {response.json()}")
except Exception as e:
    print(f"Health check failed: {e}")

print("\n" + "="*50 + "\n")

# Test 2: Brainstorm endpoint
try:
    data = {
        "product": "Makeup Tool",
        "persona": "Student"
    }
    response = requests.post('http://localhost:8000/api/brainstorm', json=data)
    print(f"Brainstorm status code: {response.status_code}")
    print(f"Brainstorm response: {response.json()}")
except Exception as e:
    print(f"Brainstorm failed: {e}")
